test/OpenSpaceToolkit/Astrodynamics/Flight/Profile/Transform.test.cpp

```diff
@@ -0,0 +1,218 @@
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_Profile_Models_Transform, IsDefined)
+{
+    {
+        EXPECT_TRUE(transform_.isDefined());
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_Profile_Models_Transform, CalculateStateAt)
+{
+    {
+        const Instant instant = Instant::DateTime(DateTime(2024, 1, 29, 0, 0, 15), Scale::UTC);
+
+        const State state = transform_.calculateStateAt(instant);
+
+        EXPECT_EQ(state.getInstant(), instant);
+
+        EXPECT_TRUE(state.accessFrame() == Frame::GCRF());
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_Profile_Models_Transform, GetAxesAt)
+{
+    // undefined
+    {
+        const Instant instant = Instant::J2000();
+        const Transform undefinedTransform = Transform::Undefined();
+        EXPECT_THROW(undefinedTransform.getAxesAt(instant), ostk::core::error::runtime::Undefined);
+    }
+
+    // instant undefined
+    {
+        const Instant instant = Instant::Undefined();
+        EXPECT_THROW(transform_.getAxesAt(instant), ostk::core::error::runtime::Undefined);
+    }
+
+    {
+        const Instant instant = Instant::DateTime(DateTime(2024, 1, 29, 0, 0, 15), Scale::UTC);
+        const Axes axes = transform_.getAxesAt(instant);
+        const State state = transform_.calculateStateAt(instant);
+        const Quaternion q_B_GCRF = state.getAttitude();
+        EXPECT_VECTORS_ALMOST_EQUAL(q_B_GCRF * axes.x(), Vector3d::X(), 1e-15);
+        EXPECT_VECTORS_ALMOST_EQUAL(q_B_GCRF * axes.y(), Vector3d::Y(), 1e-15);
+        EXPECT_VECTORS_ALMOST_EQUAL(q_B_GCRF * axes.z(), Vector3d::Z(), 1e-15);
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_Profile_Models_Transform, GetBodyFrame)
+{
+    // name undefined
+    {
+        const String frameName = String::Empty();
+        EXPECT_THROW(transform_.getBodyFrame(frameName), ostk::core::error::runtime::Undefined);
+    }
+
+    {
+        const String frameName = "OpenSpaceToolkit_Astrodynamics_Flight_Profile_Models_Transform.GetBodyFrame";
+
+        // undefined model
+        {
+            const Transform undefinedTransform = Transform::Undefined();
+            EXPECT_THROW(undefinedTransform.getBodyFrame(frameName), ostk::core::error::runtime::Undefined);
+        }
+
+        {
+            const Shared<const Frame> bodyFrame = transform_.getBodyFrame(frameName);
+            EXPECT_EQ(bodyFrame->getName(), frameName);
+        }
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_Profile_Models_Transform, Undefined)
+{
+    {
+        EXPECT_FALSE(Transform::Undefined().isDefined());
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_Profile_Models_Transform, InertialPointing)
+{
+    {
+        EXPECT_TRUE(Transform::InertialPointing(orbit_, Quaternion::Unit()).isDefined());
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Flight_Profile_Models_Transform, NadirPointing)
+{
+    {
+        EXPECT_TRUE(Transform::NadirPointing(orbit_, Orbit::FrameType::VVLH).isDefined());
+    }
+}
```

Comment:
I think you should assert a couple more things here, like the type of transform or something to verify that you actually are inertially pointing / nadir pointing with this transform

Reply:
Those are asserted in the Profile tests. Eventually we can move those tests from Profile to here.